test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,186 @@
+  let db: Db;
+  let collection: Collection;
+  let commandStartedPromise: Promise<CommandStartedEvent[]>;
+  const ops = [
+    {
+      name: 'deleteOne',
+      op: async (explain: boolean | string) => await collection.deleteOne({ a: 1 }, { explain })
+    },
+    {
+      name: 'deleteMany',
+      op: async (explain: boolean | string) => await collection.deleteMany({ a: 1 }, { explain })
+    },
+    {
+      name: 'updateOne',
+      op: async (explain: boolean | string) =>
+        await collection.updateOne({ a: 1 }, { $inc: { a: 2 } }, { explain })
+    },
+    {
+      name: 'updateMany',
+      op: async (explain: boolean | string) =>
+        await collection.updateMany({ a: 1 }, { $inc: { a: 2 } }, { explain })
+    },
+    {
+      name: 'distinct',
+      op: async (explain: boolean | string) => await collection.distinct('a', {}, { explain })
+    },
+    {
+      name: 'findOneAndDelete',
+      op: async (explain: boolean | string) =>
+        await collection.findOneAndDelete({ a: 1 }, { explain })
+    },
+    {
+      name: 'findOne',
+      op: async (explain: boolean | string) => await collection.findOne({ a: 1 }, { explain })
+    },
+    { name: 'find', op: (explain: boolean | string) => collection.find({ a: 1 }).explain(explain) },
+    {
+      name: 'findOneAndReplace',
+      op: async (explain: boolean | string) =>
+        await collection.findOneAndReplace({ a: 1 }, { a: 2 }, { explain })
+    },
+    {
+      name: 'aggregate',
+      op: async (explain: boolean | string) =>
+        await collection
+          .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }], { explain })
+          .toArray()
+    }
+  ];
+
+  beforeEach(async function () {
+    client = this.configuration.newClient({ monitorCommands: true });
+    db = client.db('queryPlannerExplainResult');
+    collection = db.collection('test');
+    await collection.insertOne({ a: 1 });
+    commandStartedPromise = once(client, 'commandStarted');
+  });
+
+  afterEach(async function () {
+    await collection.drop();
+    await client.close();
+  });
+
+  for (const explainValue of explain) {
+    context(`When explain is ${explainValue}`, function () {
+      for (const op of ops) {
+        const name = op.name;
+        it(`${name} returns ${explainValueToExpectation(explainValue)}`, async function () {
+          const response = await op.op(explainValue).catch(error => error);
+          const commandStartedEvent = await commandStartedPromise;
+          switch (explainValue) {
+            case true:
+            case 'allPlansExecution':
+              expect(commandStartedEvent[0].command.verbosity).to.be.equal('allPlansExecution');
+              if (name === 'aggregate') {
+                if (response.stages) {
+                  if (response.stages[0].$cursor) {
+                    expect(response.stages[0].$cursor).to.have.property('queryPlanner');
+                    expect(response.stages[0].$cursor).nested.property(
```

Comment:
you can eliminate the duplication of the expects throughout by conditionally setting a `responsePropertyToCheck` based on the various conditions and then just writing the expectation block once (`responsePropertyToCheck` set to `response`, `response.stages`, `response[0]`, etc); you may be able to even use a conditional `? x : y` assignment, whatever looks prettier